src/foundry/foundry.js/applications/hotbar.d.ts

```diff
@@ -1,152 +1,163 @@
+     * Get the Array of Macro (or null) values that should be displayed on a numbered page of the bar
+     * @param page -
+     */
+    protected _getMacrosByPage(page: number): Macro[];
+
+    /**
+     * Collapse the Hotbar, minimizing its display.
+     * @returns A promise which resolves once the collapse animation completes
+     */
+    collapse(): Promise<boolean>;
+
+    /**
+     * Expand the Hotbar, displaying it normally.
+     * @returns A promise which resolves once the expand animation completes
+     */
+    expand(): Promise<boolean>;
+
+    /**
+     * Change to a specific numbered page from 1 to 5
+     * @param page - The page number to change to.
+     *               (default: `1` )
+     */
+    changePage(page?: number): void;
+
+    /**
+     * Change the page of the hotbar by cycling up (positive) or down (negative)
+     * @param direction - The direction to cycle
+     */
+    cyclePage(direction?: number): void;
+
+    /** @override */
+    activateListeners(html: JQuery): void;
+
+    /**
+     * Create a Context Menu attached to each Macro button
+     * @param html - The HTML being rendered for the hotbar
+     */
+    protected _contextMenu(html: JQuery): void;
+
+    /**
+     * Handle left-click events to
+     * @param event - The originating click event
+     */
+    protected _onClickMacro(event: JQuery.ClickEvent): Promise<void>;
+
+    /**
+     * Handle hover events on a macro button to track which slot is the hover target
+     * @param event - The originating mouseover or mouseleave event
+     */
+    protected _onHoverMacro(event: JQuery.MouseEnterEvent | JQuery.MouseLeaveEvent): void;
+
+    /**
+     * Handle pagination controls
+     * @param event - The originating click event
+     */
+    protected _onClickPageControl(event: JQuery.ClickEvent): void;
+
+    /**
+     * @override
+     * @param selector - (unused)
+     */
+    protected _canDragStart(selector: string): boolean;
+
+    /** @override */
+    protected _onDragStart(event: DragEvent): false | void;
+
+    /**
+     * @override
+     * @param selector - (unused)
+     */
+    protected _canDragDrop(selector: string): boolean;
+
+    /** @override */
+    protected _onDrop(
+      event: DragEvent
+    ): void | ReturnType<InstanceType<ConfiguredDocumentClass<typeof User>>['assignHotbarMacro']>;
+
+    /**
+     * Handle click events to toggle display of the macro bar
+     * @param event -
+     */
+    protected _onToggleBar(event: JQuery.ClickEvent): void;
   }
 
-  interface DropData {
-    type?: string;
-    data?: DeepPartial<foundry.data.MacroData>; // TODO: Might be incorrect, revisit this!!
-    id?: string;
-    pack?: string;
-    slot?: number;
+  namespace Hotbar {
+    interface Data {
+      page: number;
+      macros: Macro[];
+      barClass: 'collapsed' | '';
+    }
+
+    interface DropData {
+      type?: string;
+      data?: DeepPartial<foundry.data.MacroData>; // TODO: Might be incorrect, revisit this!!
```

Comment:
This is definitely incorrect. I think the whole `DropData` type should probably just taken from the `DropData` type from `ClientDocumentMixin`: `DropData<Macro>`. Also it isn't used here at all, only in `hooks.d.ts`, so maybe just get rid of it here and just use `DropData<Macro>` there?

Also there is #928 to properly address all of the drag drop types.